### ISSUE TITLE
Fit preprocessor just once with `tune_bayes`

### ISSUE THREAD

**Issue by asb2111** [tidymodels/tune#955]
## Feature

Currently, it appears that `tune_bayes` recomputes the entire preprocessor during every iteration, even if the preprocessor has nothing to tune. This can lead to a substantial amount of unnecessary computation as the preprocessor should only need to be executed once and could be reused for all iterations.

**Comment by topepo** [comment]
This is an excellent point. 
 
Once a candidate is created by the Gaussian process model, we pass that to `tune_grid()`. It fits the new candidate for each resample, makes appropriate predictions, and gets metrics. 

We would have to make substantial changes to `tune_grid()` to skip the preprocessor. Also, we’d also have to have it save each of the fitted models from the previous fit (assuming that the preprocessor is the same), take that as input, and then start the process when the supervised model is trained (for each resample). 

We’ll have to consider this to see if there is a less invasive approach than the one described above (I don't think we can do that).

**Comment by asb2111** [comment]
If there is nothing to tune in the preprocessor, could it be 'baked' prior to starting the tuning process altogether?

**Comment by topepo** [comment]
Unless you are using a validation set, we would not want to fit the preprocessor on the entire training set then fit the model on a potentially different data set (i.e., one that was a resample)

**Comment by asb2111** [comment]
I'm thinking if we are passing in resamples we could bake the preprocessor on each resample in advance, or something like that.
